<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1170</width>
    <height>765</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Particle Swarm Optimization</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="spacing">
     <number>2</number>
    </property>
    <property name="leftMargin">
     <number>2</number>
    </property>
    <property name="topMargin">
     <number>2</number>
    </property>
    <property name="rightMargin">
     <number>2</number>
    </property>
    <property name="bottomMargin">
     <number>2</number>
    </property>
    <item>
     <widget class="QScrollArea" name="scrollArea_2">
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="scrollAreaWidgetContents_3">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>812</width>
         <height>689</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="gridLayout_4">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item row="0" column="0">
         <widget class="Canvas" name="widget" native="true">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <property name="font">
           <font>
            <stylestrategy>PreferAntialias</stylestrategy>
           </font>
          </property>
          <property name="mouseTracking">
           <bool>true</bool>
          </property>
          <property name="accessibleDescription">
           <string/>
          </property>
          <property name="autoFillBackground">
           <bool>true</bool>
          </property>
          <property name="styleSheet">
           <string notr="true"/>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="maximumSize">
       <size>
        <width>350</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="TabConfig" name="tab_config">
       <attribute name="title">
        <string>Configuration</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="0">
         <widget class="QScrollArea" name="scrollArea_3">
          <property name="horizontalScrollBarPolicy">
           <enum>Qt::ScrollBarAsNeeded</enum>
          </property>
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents_4">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>326</width>
             <height>643</height>
            </rect>
           </property>
           <layout class="QGridLayout" name="gridLayout_5">
            <item row="3" column="0">
             <widget class="QGroupBox" name="groupBox_3">
              <property name="title">
               <string>Computation</string>
              </property>
              <layout class="QVBoxLayout" name="verticalLayout_5">
               <item>
                <layout class="QFormLayout" name="formLayout_2">
                 <property name="fieldGrowthPolicy">
                  <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
                 </property>
                 <item row="2" column="0">
                  <widget class="QLabel" name="label_13">
                   <property name="minimumSize">
                    <size>
                     <width>80</width>
                     <height>0</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>Current iteration:</string>
                   </property>
                  </widget>
                 </item>
                 <item row="2" column="1">
                  <widget class="QLabel" name="label_iteration">
                   <property name="text">
                    <string/>
                   </property>
                  </widget>
                 </item>
                 <item row="3" column="0">
                  <widget class="QLabel" name="label_5">
                   <property name="minimumSize">
                    <size>
                     <width>80</width>
                     <height>0</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>Max iterations:</string>
                   </property>
                  </widget>
                 </item>
                 <item row="3" column="1">
                  <widget class="QLineEdit" name="lineEdit_maxIterations">
                   <property name="inputMask">
                    <string/>
                   </property>
                   <property name="text">
                    <string/>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QGroupBox" name="groupBox_2">
              <property name="title">
               <string>Optimization</string>
              </property>
              <layout class="QVBoxLayout" name="verticalLayout_2">
               <item>
                <layout class="QFormLayout" name="formLayout">
                 <property name="fieldGrowthPolicy">
                  <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
                 </property>
                 <item row="6" column="0">
                  <widget class="QLabel" name="label_2">
                   <property name="minimumSize">
                    <size>
                     <width>80</width>
                     <height>0</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>Cognitive factor:</string>
                   </property>
                  </widget>
                 </item>
                 <item row="6" column="1">
                  <widget class="QLineEdit" name="lineEdit_cp"/>
                 </item>
                 <item row="7" column="0">
                  <widget class="QLabel" name="label_3">
                   <property name="minimumSize">
                    <size>
                     <width>80</width>
                     <height>0</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>Social factor:</string>
                   </property>
                  </widget>
                 </item>
                 <item row="7" column="1">
                  <widget class="QLineEdit" name="lineEdit_cg"/>
                 </item>
                 <item row="8" column="0">
                  <widget class="QLabel" name="label_4">
                   <property name="minimumSize">
                    <size>
                     <width>80</width>
                     <height>0</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>Inertia factor:</string>
                   </property>
                  </widget>
                 </item>
                 <item row="8" column="1">
                  <widget class="QLineEdit" name="lineEdit_omega"/>
                 </item>
                 <item row="5" column="0">
                  <widget class="QLabel" name="label_11">
                   <property name="minimumSize">
                    <size>
                     <width>80</width>
                     <height>0</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>Total particles</string>
                   </property>
                  </widget>
                 </item>
                 <item row="5" column="1">
                  <widget class="QLabel" name="label_particles_number">
                   <property name="font">
                    <font>
                     <weight>50</weight>
                     <italic>false</italic>
                     <bold>false</bold>
                    </font>
                   </property>
                   <property name="text">
                    <string/>
                   </property>
                  </widget>
                 </item>
                 <item row="9" column="0">
                  <widget class="QLabel" name="label_14">
                   <property name="minimumSize">
                    <size>
                     <width>80</width>
                     <height>0</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>Max initial velocity:</string>
                   </property>
                  </widget>
                 </item>
                 <item row="9" column="1">
                  <widget class="QLineEdit" name="lineEdit_maxVelocity"/>
                 </item>
                </layout>
               </item>
              </layout>
             </widget>
            </item>
            <item row="1" column="0">
             <layout class="QVBoxLayout" name="verticalLayout_3">
              <item>
               <widget class="QGroupBox" name="groupBox_4">
                <property name="title">
                 <string>Fitness function</string>
                </property>
                <layout class="QVBoxLayout" name="verticalLayout_4">
                 <item>
                  <widget class="QComboBox" name="comboBox"/>
                 </item>
                </layout>
               </widget>
              </item>
             </layout>
            </item>
            <item row="7" column="0">
             <spacer name="verticalSpacer">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
            <item row="4" column="0">
             <widget class="QGroupBox" name="groupBox_7">
              <property name="title">
               <string>Solution</string>
              </property>
              <layout class="QGridLayout" name="gridLayout_10">
               <item row="0" column="0" colspan="2">
                <layout class="QFormLayout" name="formLayout_8">
                 <item row="0" column="0">
                  <widget class="QLabel" name="label_12">
                   <property name="minimumSize">
                    <size>
                     <width>80</width>
                     <height>0</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>Best fitness:</string>
                   </property>
                  </widget>
                 </item>
                 <item row="0" column="1">
                  <widget class="QLabel" name="label_best_fitness">
                   <property name="text">
                    <string/>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="0">
                  <widget class="QLabel" name="label_15">
                   <property name="minimumSize">
                    <size>
                     <width>80</width>
                     <height>0</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>Best solution:</string>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="1">
                  <widget class="QLabel" name="label_best_position">
                   <property name="text">
                    <string/>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </widget>
            </item>
            <item row="8" column="0">
             <layout class="QVBoxLayout" name="verticalLayout">
              <item>
               <widget class="QPushButton" name="button_default">
                <property name="text">
                 <string>Set default values</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="TabParticle" name="tab_particle">
       <attribute name="title">
        <string>Particle</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="0" column="0">
         <widget class="QScrollArea" name="scrollArea_4">
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents_2">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>326</width>
             <height>643</height>
            </rect>
           </property>
           <layout class="QGridLayout" name="gridLayout_6">
            <item row="1" column="0">
             <widget class="QGroupBox" name="groupBox">
              <property name="title">
               <string>Position</string>
              </property>
              <layout class="QGridLayout" name="gridLayout_7">
               <item row="0" column="0">
                <layout class="QFormLayout" name="formLayout_5">
                 <item row="1" column="0">
                  <widget class="QLabel" name="label">
                   <property name="minimumSize">
                    <size>
                     <width>80</width>
                     <height>0</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>Actual:</string>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="1">
                  <widget class="QLabel" name="label_particle_position">
                   <property name="font">
                    <font>
                     <weight>75</weight>
                     <bold>true</bold>
                    </font>
                   </property>
                   <property name="styleSheet">
                    <string notr="true"/>
                   </property>
                   <property name="text">
                    <string/>
                   </property>
                  </widget>
                 </item>
                 <item row="2" column="0">
                  <widget class="QLabel" name="label_6">
                   <property name="minimumSize">
                    <size>
                     <width>80</width>
                     <height>0</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>Next:</string>
                   </property>
                  </widget>
                 </item>
                 <item row="2" column="1">
                  <widget class="QLabel" name="label_particle_next_position">
                   <property name="text">
                    <string/>
                   </property>
                  </widget>
                 </item>
                 <item row="3" column="0">
                  <widget class="QLabel" name="label_9">
                   <property name="minimumSize">
                    <size>
                     <width>80</width>
                     <height>0</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>Best:</string>
                   </property>
                  </widget>
                 </item>
                 <item row="3" column="1">
                  <widget class="QLabel" name="label_particle_best_position">
                   <property name="text">
                    <string/>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QGroupBox" name="groupBox_5">
              <property name="title">
               <string>Fitness</string>
              </property>
              <layout class="QGridLayout" name="gridLayout_8">
               <item row="0" column="0">
                <layout class="QFormLayout" name="formLayout_6">
                 <item row="0" column="0">
                  <widget class="QLabel" name="label_8">
                   <property name="minimumSize">
                    <size>
                     <width>80</width>
                     <height>0</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>Actual:</string>
                   </property>
                  </widget>
                 </item>
                 <item row="0" column="1">
                  <widget class="QLabel" name="label_particle_fitness">
                   <property name="font">
                    <font>
                     <weight>75</weight>
                     <bold>true</bold>
                    </font>
                   </property>
                   <property name="text">
                    <string/>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="0">
                  <widget class="QLabel" name="label_17">
                   <property name="minimumSize">
                    <size>
                     <width>80</width>
                     <height>0</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>Best:</string>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="1">
                  <widget class="QLabel" name="label_particle_best_fitness">
                   <property name="text">
                    <string/>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QGroupBox" name="groupBox_6">
              <property name="title">
               <string>Velocity:</string>
              </property>
              <layout class="QGridLayout" name="gridLayout_9">
               <item row="2" column="0">
                <layout class="QFormLayout" name="formLayout_7">
                 <property name="fieldGrowthPolicy">
                  <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
                 </property>
                 <item row="1" column="0">
                  <widget class="QLabel" name="label_7">
                   <property name="minimumSize">
                    <size>
                     <width>80</width>
                     <height>0</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>Actual:</string>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="1">
                  <widget class="QLabel" name="label_particle_velocity">
                   <property name="font">
                    <font>
                     <weight>75</weight>
                     <bold>true</bold>
                    </font>
                   </property>
                   <property name="text">
                    <string/>
                   </property>
                  </widget>
                 </item>
                 <item row="5" column="0">
                  <widget class="QLabel" name="label_10">
                   <property name="minimumSize">
                    <size>
                     <width>80</width>
                     <height>0</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>Inert:</string>
                   </property>
                  </widget>
                 </item>
                 <item row="5" column="1">
                  <widget class="QLabel" name="label_particle_vi">
                   <property name="text">
                    <string/>
                   </property>
                  </widget>
                 </item>
                 <item row="2" column="1">
                  <widget class="Line" name="line">
                   <property name="lineWidth">
                    <number>0</number>
                   </property>
                   <property name="orientation">
                    <enum>Qt::Horizontal</enum>
                   </property>
                  </widget>
                 </item>
                 <item row="3" column="0">
                  <widget class="QLabel" name="label_19">
                   <property name="minimumSize">
                    <size>
                     <width>80</width>
                     <height>0</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>Social:</string>
                   </property>
                  </widget>
                 </item>
                 <item row="3" column="1">
                  <widget class="QLabel" name="label_particle_vg">
                   <property name="text">
                    <string/>
                   </property>
                  </widget>
                 </item>
                 <item row="4" column="0">
                  <widget class="QLabel" name="label_18">
                   <property name="minimumSize">
                    <size>
                     <width>80</width>
                     <height>0</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>Cognitive:</string>
                   </property>
                  </widget>
                 </item>
                 <item row="4" column="1">
                  <widget class="QLabel" name="label_particle_vp">
                   <property name="styleSheet">
                    <string notr="true"/>
                   </property>
                   <property name="text">
                    <string/>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </widget>
            </item>
            <item row="3" column="0">
             <spacer name="verticalSpacer_2">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionRandom"/>
   <addaction name="actionClear"/>
   <addaction name="separator"/>
   <addaction name="actionStep"/>
   <addaction name="actionRun"/>
   <addaction name="actionStop"/>
   <addaction name="actionFinish"/>
   <addaction name="actionRestart"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1170</width>
     <height>19</height>
    </rect>
   </property>
  </widget>
  <action name="actionRandom">
   <property name="text">
    <string>Random</string>
   </property>
   <property name="toolTip">
    <string>Generate 10 random particles.</string>
   </property>
  </action>
  <action name="actionClear">
   <property name="text">
    <string>Clear</string>
   </property>
   <property name="toolTip">
    <string>Remove particles.</string>
   </property>
  </action>
  <action name="actionStep">
   <property name="text">
    <string>Step</string>
   </property>
   <property name="toolTip">
    <string>Run one iteration.</string>
   </property>
  </action>
  <action name="actionFinish">
   <property name="text">
    <string>Finish</string>
   </property>
   <property name="toolTip">
    <string>Compute the optimum.</string>
   </property>
  </action>
  <action name="actionRun">
   <property name="text">
    <string>Run</string>
   </property>
   <property name="toolTip">
    <string>Animate the optimization.</string>
   </property>
  </action>
  <action name="actionStop">
   <property name="text">
    <string>Stop</string>
   </property>
   <property name="toolTip">
    <string>Stop the optimization.</string>
   </property>
  </action>
  <action name="actionRestart">
   <property name="text">
    <string>Restart</string>
   </property>
   <property name="toolTip">
    <string>Restart optimization.</string>
   </property>
  </action>
  <action name="actionDefault">
   <property name="text">
    <string>Default</string>
   </property>
   <property name="toolTip">
    <string>Set default values.</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>Canvas</class>
   <extends>QWidget</extends>
   <header>canvas.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>TabParticle</class>
   <extends>QWidget</extends>
   <header>tabparticle.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>TabConfig</class>
   <extends>QWidget</extends>
   <header>tabconfig.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
